nnfabrik/templates/checkpoint.py

```diff
@@ -0,0 +1,166 @@
+import copy
+import os
+import tempfile
+
+import datajoint as dj
+import torch
+
+from nnfabrik.templates.trained_model import TrainedModelBase
+from nnfabrik.utility.dj_helpers import make_hash, clone_conn, CustomSchema
+
+
+def my_checkpoint(nnfabrik):
+    conn_clone = clone_conn(dj.conn())
+    schema_clone = CustomSchema(nnfabrik.schema.database, connection=conn_clone)
+
+    @schema_clone
+    class Checkpoint(dj.Manual):
+        storage = "minio"
+
+        @property
+        def definition(self):
+            definition = f"""
+            # Checkpoint table
+            -> nnfabrik.Trainer
+            -> nnfabrik.Dataset
+            -> nnfabrik.Model
+            -> nnfabrik.Seed
+            epoch:                             int          # epoch of creation
+            ---
+            score:                             float        # current score at epoch
+            state:                             attach@{self.storage}  # current state
+            ->[nullable] nnfabrik.Fabrikant
+            trainedmodel_ts=CURRENT_TIMESTAMP: timestamp    # UTZ timestamp at time of insertion
+            """
+            return definition
+
+    return Checkpoint
+
+
+class TrainedModelChkptBase(TrainedModelBase):
+    checkpoint_table = None  # Checkpoint
+    keys = [
+        "model_fn",
+        "model_hash",
+        "dataset_fn",
+        "dataset_hash",
+        "trainer_fn",
+        "trainer_hash",
+    ]
+
+    def call_back(self, uid=None, epoch=None, model=None, state=None):
```

Comment:
This overwrites the `call_back` method in `TrainedModelBase`. Since the `call_back` there is designed to be quite flexible, I believe the name has to stay.

Reply:
Not ideal but I guess nothing you can do about it in the context of this pull request.